crates/store/re_redap_client/src/grpc.rs

```diff
@@ -517,39 +496,55 @@ async fn stream_partition_from_server(
     let store_id = store_info.store_id.clone();
 
     if tx
-        .send(LogMsg::SetStoreInfo(SetStoreInfo {
-            row_id: *re_chunk::RowId::new(),
-            info: store_info,
-        }))
+        .send(
+            LogMsg::SetStoreInfo(SetStoreInfo {
+                row_id: *re_chunk::RowId::new(),
+                info: store_info,
+            })
+            .into(),
+        )
         .is_err()
     {
         re_log::debug!("Receiver disconnected");
         return Ok(());
     }
 
     // Send UI commands for recording (as opposed to blueprint) stores.
-    if let Some(on_ui_cmd) = on_ui_cmd
-        && store_id.is_recording()
-    {
-        if let Some(time_range) = time_range {
-            on_ui_cmd(UiCommand::AddValidTimeRange {
+    if store_id.is_recording() {
+        let valid_range_msg = if let Some(time_range) = time_range {
+            DataSourceUiCommand::AddValidTimeRange {
                 store_id: store_id.clone(),
                 timeline: Some(*time_range.timeline.name()),
                 time_range: time_range.into(),
-            });
+            }
         } else {
-            on_ui_cmd(UiCommand::AddValidTimeRange {
+            DataSourceUiCommand::AddValidTimeRange {
                 store_id: store_id.clone(),
                 timeline: None,
                 time_range: AbsoluteTimeRange::EVERYTHING,
-            });
+            }
+        };
+
+        if tx.send(valid_range_msg.into()).is_err() {
+            re_log::debug!("Receiver disconnected");
+            return Ok(());
         }
 
+        #[expect(clippy::collapsible_if)]
         if !fragment.is_empty() {
-            on_ui_cmd(UiCommand::SetUrlFragment {
-                store_id: store_id.clone(),
-                fragment,
-            });
+            if tx
+                .send(
+                    DataSourceUiCommand::SetUrlFragment {
+                        store_id: store_id.clone(),
+                        fragment: fragment.to_string(),
+                    }
+                    .into(),
+                )
```

Comment:
style: we could make `send` take an `impl Into<…>` 🤷 

```suggestion
                .send(
                    DataSourceUiCommand::SetUrlFragment {
                        store_id: store_id.clone(),
                        fragment: fragment.to_string(),
                    }
                )
```